Hazel/Dtls/Handshake.cs

```diff
@@ -481,9 +481,12 @@ public static ByteSpan ComputeAddressMac(EndPoint peerAddress, HMAC hmac)
                 data[ii] = address[ii];
             }
 
-            ///NOTE(mendsley): Lame that we need to allocate+copy here
-            ByteSpan signature = hmac.ComputeHash(data);
-            return signature.Slice(0, CookieSize);
+            lock (hmac)
```

Comment:
Curious why the lock is necessary here---is there some danger when the current hmac gets rotated? or something internal to hmac prevents it from being used concurrently?

Also, possibly not directly related--it looks like the hmac rotation might also need protection. That might be a good candidate to pull out of the message processing pathway and on its own timer

Reply:
I found that if I spammed the AmongUsServer with valid connection data (but no app data after) this hmac would throw an InvalidAccessException. This is because we're locking at the peer level, but this hmac exists at the Listener level and it isn't thread safe. Before it was only used by one thread, so it was safe.

Reply:
Got it! I'll see if there's something we can do